packages/pulumi-aws/src/apps/gateway/GatewayLambdas.ts

```diff
@@ -0,0 +1,82 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+
+import { PulumiApp } from "@webiny/pulumi-sdk";
+import { buildCloudFrontFunction, buildLambdaEdge } from "@webiny/project-utils";
+
+export function createLambdas(app: PulumiApp) {
+    const role = app.addResource(aws.iam.Role, {
+        name: `lambda-edge-role`,
+        config: {
+            managedPolicyArns: [aws.iam.ManagedPolicies.AWSLambdaBasicExecutionRole],
+            assumeRolePolicy: {
+                Version: "2012-10-17",
+                Statement: [
+                    {
+                        Action: "sts:AssumeRole",
+                        Principal: aws.iam.Principals.LambdaPrincipal,
+                        Effect: "Allow"
+                    },
+                    {
+                        Action: "sts:AssumeRole",
+                        Principal: aws.iam.Principals.EdgeLambdaPrincipal,
+                        Effect: "Allow"
+                    }
+                ]
+            }
+        }
+    });
+
+    const functions = app.addHandler(() => {
```

Comment:
Can you clarify a bit...

When do we use the `addHandler`? And can you describe what it does briefly?

Reply:
Sure, I'll add more comments.
But it's difficult to understand the whole concept just from comments. maybe I will have to write some notion doc, to describe how new pulumi architecture works.